lib/ansible/module_utils/facts/system/aix.py

```diff
@@ -0,0 +1,257 @@
+except ImportError:
+    import simplejson as json
+
+
+from ansible.module_utils.facts.utils import get_file_content
+
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+
+
+class AIXSystemFactCollector(BaseFactCollector):
+    """
+    Some handy AIX facts, more or less nice to haves
+    """
+    name = 'aix_system_facts'
+    _fact_ids = set()
+    _platform = 'AIX'
+
+    def get_lssrc_facts(self, module=None, collected_facts=None):
+        """
+        'lssrc': [
+                  {
+                   'Subsystem': 'prngd',
+                   'PID': '3211316',
+                   'Group': 'prng',
+                   'Status': 'active'
+                   },
+                  {
+                   'Subsystem': 'syslogd',
+                   'PID': '4849686',
+                   'Group': 'ras',
+                   'Status': 'active'
+                   }
+                   ],
+        """
+        lssrc_facts = {}
+
+        lssrc_facts['lssrc'] = []
+
+        lssrc = []
+
+        rc, out, err = self.module.run_command(["/usr/bin/lssrc", "-a"])
+        if rc != 0:
+            self.module.fail_json(
+                msg="ERROR: Could not run lssrc command",
+                rc=rc,
+                err=err)
+        firstline = True
+        for line in out.splitlines():
+            for line in out.splitlines():
+                if firstline:
+                    keys = line.split()
+                    firstline = False
+                else:
+                    values = [line[0:18].strip(), line[18:34].strip(),
+                              line[34:48].strip(), line[48:60].strip()]
+                    adict = dict(itertools.izip(keys, values))
+                    lssrc.append(adict)
+        lssrc_facts['lssrc'] = lssrc
+
+        return lssrc_facts
+
+    def get_niminfo_facts(self, module=None, collected_facts=None):
+        """
+        {'niminfo': [
+                     {
+                      'NIM_BOS_FORMAT': 'rte',
+                      'NIM_CONFIGURATION': 'standalone',
+                      'NIM_MOUNTS': '',
+                      'NIM_MASTERID': '00FB1A6B4C00',
+                      'NIM_NAME': 'aixbuildhostng',
+                      'NIM_MASTER_HOSTNAME': 'nimmhs',
+                      'NIM_HOSTNAME': 'aixbuildhostng',
+                      'NIM_SHELL': 'nimsh',
+                      'NIM_MASTER_PORT': '1058',
+                      'NIM_HOSTS': '127.0.0.1:loopback:localhost  X.X.14.50:aixbuildhostng  X.X.14.211:nimmhs',
+                      'NIM_REGISTRATION_PORT': '1059',
+                      'NIM_FIPS_MODE': '0',
+                      'NIM_BOS_IMAGE': '/SPOT/usr/sys/inst.images/installp/ppc/bos'
+                      }
+                      ],
+        """
+        file = '/etc/niminfo'
+        niminfo_facts = {}
+
+        niminfo_facts['niminfo'] = []
+
+        niminfo = []
+
+        try:
+            if os.path.exists(file):
+                niminfo = dict(
```

Comment:
Can we reduce the complexity of this block by splitting it?
Also put the keys in lowercase